torch_xla/csrc/init_python_bindings.cpp

```diff
@@ -408,15 +408,64 @@ std::ptrdiff_t GetTensorId(const at::Tensor& tensor) {
   return xtensor->GetUniqueId();
 }
 
+// Converts a py::list of torch_xla.experimental.xla_sharding.ShardingSpec
+// into a vector of xla::OpSharding that can be passed into CreateTensorsData
+// along with the tensors to create sharded XLA tensors. A value of None in
+// the input list corresponds to no sharding on the tensor, which in SPMD mode
+// results in implicit replication of the tensor across devices.
+//
+// If the input `py_sharding_specs` is std::nullopt, then the return value is
+// also std::nullopt.
+std::optional<std::vector<XLATensor::ShardingSpecPtr>> ShardingSpecFromPyHandle(
+    const std::optional<py::list>& py_sharding_specs) {
+  static const py::object pyShardingSpec =
+      py::module_::import("torch_xla.experimental.xla_sharding")
```

Comment:
Can we define a proper PYBIND11_MODULE for sharding spec? Then I guess you don't need this helper.

Reply:
Okay, I guess we have two different ShardingSpec type in python and c++ such that you need a conversion. What's the differences? Can we merge them?

Reply:
I don't think we'll be able to merge them, the ShardingSpec in python provides a lightweight wrapper around the Mesh and PartitionSpec while the xla::OpSharding is the proto passed to the XLA compiler.

I like your idea though, I can define a pybind class here wrapping the `xla::OpSharding` to support the conversion in the python layer. I'll try that out, thanks Jiewen!

Reply:
+1 @alanwaketan for defining a proper module in Pybind11 for "Python" `ShardingSpec`, I think what you are describing @jonb377 could make things cleaner.

Also, just to add some more context -- the python `ShardingSpec` is to represent sharding info for the data loader and the `mark_sharding` API, while the C++ `ShardingSpec` is a wrapper around `xla::OpSharding` proto for the compiler to work with. 

Some naming conflicts there, but the user should only care about the python one, and how they want to think about it wrt the `mark_sharding` API. The proto should be transparent.

Reply:
Thanks for the clarification. I guess maybe we can rename one of them later as a follow up to reduce confusions.